src/Datadog.Trace.ClrProfiler.Managed/AutoInstrumentation/AWS/SQS/SendMessageAsyncIntegration.cs

```diff
@@ -0,0 +1,72 @@
+using System;
+using System.Threading;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Tagging;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AWS.SQS
+{
+    /// <summary>
+    /// AWSSDK.SQS SendMessageAsync calltarget instrumentation
+    /// </summary>
+    [InstrumentMethod(
+        AssemblyName = "AWSSDK.SQS",
+        TypeName = "Amazon.SQS.AmazonSQSClient",
+        MethodName = "SendMessageAsync",
+        ReturnTypeName = "System.Threading.Tasks.Task`1<Amazon.SQS.Model.SendMessageResponse>",
+        ParameterTypeNames = new[] { "Amazon.SQS.Model.SendMessageRequest", ClrNames.CancellationToken },
+        MinimumVersion = "3.0.0",
+        MaximumVersion = "3.*.*",
+        IntegrationName = AwsConstants.IntegrationName)]
+    public class SendMessageAsyncIntegration
+    {
+        private const string Operation = "SendMessage";
+
+        /// <summary>
+        /// OnMethodBegin callback
+        /// </summary>
+        /// <typeparam name="TTarget">Type of the target</typeparam>
+        /// <typeparam name="TSendMessageRequest">Type of the request object</typeparam>
+        /// <param name="instance">Instance value, aka `this` of the instrumented method</param>
+        /// <param name="request">The request for the SQS operation</param>
+        /// <param name="cancellationToken">CancellationToken value</param>
+        /// <returns>Calltarget state value</returns>
+        public static CallTargetState OnMethodBegin<TTarget, TSendMessageRequest>(TTarget instance, TSendMessageRequest request, CancellationToken cancellationToken)
+        {
+            if (request is null)
+            {
+                return CallTargetState.GetDefault();
+            }
+
+            var requestProxy = request.DuckCast<ISendMessageRequest>();
+
+            var scope = AwsSqsCommon.CreateScope(Tracer.Instance, $"{AwsConstants.AwsService}.{Operation}", out AwsSqsTags tags);
+            tags.Operation = Operation;
+            tags.Service = AwsConstants.AwsService;
+            tags.QueueUrl = requestProxy.QueueUrl;
+
+            if (scope?.Span?.Context != null)
```

Comment:
Is there a case here where `scope!=null` but `scope.Span==null` 🤔

Reply:
Yeah you're right, in this automatic instrumentation I know that if I created a Scope then there is a Span associated with it. I'll remove the additional null-check here. Fixed in 9a9e2a736689c1cf169919587ae856e406d8c188